### PR TITLE
if no deployments are found, we can't deploy

### PR DESCRIPTION
**Pull request by sjmiller609**
```
// Deploy pushes a new docker image
func Deploy(path, name, wsId string, prompt bool) error {

        ... omitted

	// Get Deployments from workspace ID
	deReq := houston.Request{
		Query:     houston.DeploymentsGetRequest,
		Variables: map[string]interface{}{"workspaceId": currentWorkspace.Id},
	}

	deResp, err := deReq.Do()
	if err != nil {
		return err
	}

	deployments := deResp.Data.GetDeployments

        ... omitted

	// NOTE: this part is working - when I do "astro deploy" (without specifying a deployment name) it does report error
	// Prompt user for deployment if no deployment passed in
	if len(name) == 0 || prompt {
		if len(deployments) == 0 {
			return errors.New(messages.HOUSTON_NO_DEPLOYMENTS_ERROR)
		}

		... omitted
	}

	// NOTE: Since we have no deployments, this will not run
        // that is why the tag is blank.
	nextTag := ""
	for _, deployment := range deployments {
		if deployment.ReleaseName == name {
			nextTag = deployment.DeploymentInfo.Next
		}
	}
        ... omitted
}
```
Now instead of this error:
```

Stevens-MacBook-Pro:airflow_dev steven$ astro deploy test
Deploying: test
test/airflow
Building image...
Sending build context to Docker daemon  13.31kB
Step 1/1 : FROM astronomerinc/ap-airflow:0.9.7-1.10.3-onbuild
# Executing 5 build triggers
 ---> Using cache
 ---> Using cache
 ---> Using cache
 ---> Using cache
 ---> Using cache
 ---> 625af0ec11fa
Successfully built 625af0ec11fa
Successfully tagged test/airflow:latest
Error parsing reference: "registry.staging.astronomer.io/test/airflow:" is not a valid repository/tag: invalid reference format
Error: command 'docker tag test/airflow:latest registry.staging.astronomer.io/test/airflow:' failed: failed to execute cmd: exit status 1
```

we get this error:
```
Stevens-MacBook-Pro:airflow_dev steven$ astro deploy test
Error: No airflow deployments found
```

listing deployments remains to be solved.

### CODE REVIEW

File: airflow/docker.go [hunk 1]
```diff
@@ -532,11 +532,12 @@ func Deploy(path, name, wsId string, prompt bool) error {
 		name = config.CFG.ProjectDeployment.GetProjectString()
 	}
 
+	if len(deployments) == 0 {
+		return errors.New(messages.HOUSTON_NO_DEPLOYMENTS_ERROR)
+	}
+
```
Comment (sjmiller609) [inline airflow/docker.go]: this doesn't fix our current problem, but it fixes the client side error message to be correct in the case of houston not returning a list of deployments. i.e. if a user had a workspace with no deployments, then tried to deploy using a name, ```astro deploy this-deployment-was-not-created``` they would find the same issue. this is an informative error message instead.

File: .gitignore [hunk 2]
```diff
@@ -9,4 +9,5 @@ requirements.txt
 # IDEs
 .idea/
 .vscode/
-.DS_Store
\ No newline at end of file
+.DS_Store
+explain.md
```
Comment (andriisoldatenko) [inline .gitignore]: what is explain.md?

**Comment by andriisoldatenko** [comment]
https://github.com/astronomer/astro-cli/issues ?

File: .gitignore (hunk 2 above)
Comment (sjmiller609) [reply .gitignore]: I can remove it. just a file I have locally

**Comment by sjmiller609** [comment]
oh what do you know? https://github.com/astronomer/astro-cli/issues/247

**Pull request merged**